lib/pure/collections/sets.nim

```diff
@@ -80,6 +80,10 @@ type
     ## <#initOrderedSet,int>`_ before calling other procs on it.
     data: OrderedKeyValuePairSeq[A]
     counter, first, last: int
+  SomeSet[T] = HashSet[T] | OrderedSet[T]
+  Iterable[T] = concept x
+    for value in x:
+      type(value) is T
```

Comment:
Don't use concepts that aren't exported, how can the user of this API use this concept?

Exporting this concept would require a separate PR (it should be added in a different module and used throughout the stdlib), so for this one you simply shouldn't use it IMO.

Reply:
without concepts, this is a recipie for introducing bloat like `proc toHashSet*[A](s: OrderedSet[A]): HashSet[A] =` (see my remark in https://github.com/nim-lang/Nim/pull/14481/files#r439715608)